tests/conftest.py

```diff
@@ -4,11 +4,11 @@
 from .implementations import *
 
 
-@pytest.fixture(params=implementations)
+@pytest.fixture(params=implementations, scope="session")
```

Comment:
```suggestion
@pytest.fixture(params=implementations, scope="function")
```
There should be a clean testing environment per test. Default is function.

Reply:
In general I agree, but later you use the same client to insert data and since in pagination test i refactored it to be `class` fixture to not insert it multiple times as you only fetch data, and fetch from the same data, in your test, the `class` scope fixture cannot depend on `function` scope one. 

Pagination fixtures can be changed to `function` scope too, but then it will be much slower, it's still much faster then before as before you run the inserts not per function but per parametrization. 

Alternative is to insert data in a different way for pagination purposes, although if client will clean the data on each run (and is function scope) it's still have to be function scope. (I'm not sure if you clean the data in client?)

Reply:
I agree. I have changed it to scope class which seems to be the best of both worlds.